File: antiguo/src/main/frontend/bookwave/src/app/models/Book.ts
import { Category } from "./Category";

export interface Book {
    id: number;
    title: string;
    author: string;
    description: string;
    cover: string;
    language: string;
    year: number;
    categories?: Category[];
    readCount?: number;
    readingLists?: Book[];
}

File: bookwave/src/main/java/bookwave/directory/bookwave/domain/Book.java
package bookwave.directory.bookwave.domain;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import java.util.Set;

@Entity
@Table(name = "books")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Book {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String title;
    private String author;
    private String description;
    private String cover;
    private String language;
    private int year;

    //Relaciones
    @ManyToMany(mappedBy = "books")
    @JsonIgnore
    private Set<Category> categories;
}

